modules/invibesBidAdapter.js

```diff
@@ -300,35 +310,69 @@ function getCappedCampaignsAsString() {
     .join(',');
 }
 
-function initLogger() {
-  const noop = function () { };
+const noop = function () { };
 
+function initLogger() {
   if (localStorage && localStorage.InvibesDEBUG) {
     return window.console;
   }
 
   return { info: noop, error: noop, log: noop, warn: noop, debug: noop };
 }
 
+function buildSyncUrl() {
+  let syncUrl = _customUserSync || CONSTANTS.SYNC_ENDPOINT;
+  syncUrl += '?visitId=' + invibes.visitId;
+
+  if (invibes.optIn) {
+    syncUrl += '&optIn=1';
+  }
+
+  const did = invibes.getCookie('ivbsdid');
+  if (did) {
+    syncUrl += '&ivbsdid=' + encodeURIComponent(did);
+  }
+
+  const bks = invibes.getCookie('ivvbks');
+  if (bks) {
+    syncUrl += '&ivvbks=' + encodeURIComponent(bks);
+  }
+
+  return syncUrl;
+}
+
+function handlePostMessage() {
+  try {
+    if (window.addEventListener) {
+      window.addEventListener('message', acceptPostMessage);
+    }
+  } catch (e) { }
+}
+
+function acceptPostMessage(e) {
```

Comment:
What's the reason the this message?  It's not clear to me why the caller does take care of setting cookies, etc and why the code to do so is inside your Prebid adapter.

Reply:
The user sync happens in the iframe, on a different domain that the initial page. We need to update our local id if the user sync was successful, so the only way to do this was with a postMessage from the iframe.